exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/metadata/MetadataAggregateHelper.java

```diff
@@ -313,8 +317,19 @@ private void addColumnAggregateCalls(FieldReference fieldRef, String fieldName)
       if (interestingColumns == null || interestingColumns.contains(fieldRef)) {
         // collect statistics for all or only interesting columns if they are specified
         AnalyzeColumnUtils.COLUMN_STATISTICS_FUNCTIONS.forEach((statisticsKind, sqlKind) -> {
+          // constructs "case when is not null projectMetadataColumn then column1 else null end" call
+          // to avoid using default values for required columns when data for empty result is obtained
```

Comment:
Just FYI: a clever thing Impala does is to use just plain SQL to gather stats. Then, logic such as this is expressed in the SQL for the stats query. I think Impala has a few functions specifically for stats gathering, such as NDV estimates, etc.

Reply:
Thanks for pointing this. Unfortunately, we can't use a plain SQL approach to collect metadata, since we do not have information about the schema, so we create aggregate calls dynamically. But Drill uses inbuilt aggregate functions for collecting summary statistics (`MIN`, `MAX`, ...).